Neha Pusarla 

Kedar Rudrawar 


Sahil Saboo 


## Rolling Circle Amplification 

Rolling Circle Amplification (RCA) is a process in which a short DNA or RNA primer is amplified to create a long single-stranded DNA or RNA using its respective polymerases and a circular DNA template. This enzymatic process proves to be a powerful DNA amplification technique that plays a significant role in biomedical research and nanotechnology. RCA is used to develop multiple sensitive diagnostic methods that have different targets such as proteins, cells, nucleic acids, and other small molecules. In addition, because the RCA products are circular in structure due to the circular DNA template, the products have been used to create complicated DNA structures such as nanotubes, nanoribbons, and other DNA materials. These specific RCA products have been used for drug delivery, bioseparation, biodetection, and other technologies. In addition, RCA is a powerful tool for amplifying signals in immunoassays, especially ELISA, and as such has become a very versatile amplification tool with a menagerie of applications in genomics, proteomics, diagnosis, biosensing, and enzyme development. 

A typical RCA experiment works through the use of five main components: DNA Polymerase (such as ɸ29 DNA Polymerase) or RNA polymerase, an applicable buffer, DNA or RNA primer, single-stranded DNA template, and dNTPs for elongation. The first step is to ligate the two ends of the single-stranded DNA into one circular template, which can either be done with a ligation primer and enzyme, such as T4 DNA ligase, or with just a template-free ligation enzyme, such as CircLigase [3], shown in Figure 1A. The target sequence is often used as the primer for extension, RNA or DNA, and can serve as a ligation primer as well. Once the primer anneals to the newly formed circular template, the polymerase will attach to the primer and elongate the sequence. While this elongation occurs, the circular template starts to rotate, or “roll”, pushing the previously existing primer and extension off the template. A derivative of RCA, known as Hyperbranched RCA (HRCA), exists as well, in which multiple primers anneal to the circular template, and multiple polymerases can attach and simultaneously extend the replicating sequence, to produce many more replicated strands (shown in Figure 1D). Variations of approaches can be shown in Figure 1B, 1C, and 1D, and further analysis paths can be shown by 1E, 1F, and 1G. 

<br />

![Figure 1 - RCA Pathways](https://github.com/npusarla/BENG183_Final_Projects_FALL2019/blob/master/Images/Figure1.png)
&nbsp; &nbsp; &nbsp; &nbsp;**Figure 1 - RCA Pathways**
<br />


One of the main applications of Rolling Circle Amplification (RCA) is as a replacement for Polymerase Chain Reaction (PCR) as the replication/amplification process of genetic material for sequencing. For example, with the normal amount of genetic material generally used to initiate Sanger sequencing, it would only take about five hours to get the necessary amount of genetic material using Rolling Circle Amplification. The pipeline for Sanger sequencing then takes about seven hours, allowing the sequencing of genetic material to have a turnaround time of one day. PCR, on the other hand, requires plasmid preparation (miniprep), which causes the total process of amplification and sequencing to last multiple days. RCA offers a huge advantage by allowing the replication process of genetic material to be performed without the preparation required, unlike PCR, the current industry standard.

<br />

![Figure 2 - PCR and RCA Pipelines Side by Side](https://github.com/npusarla/BENG183_Final_Projects_FALL2019/blob/master/Images/Figure2.png)
&nbsp; &nbsp; &nbsp; &nbsp;**Figure 2 - PCR and RCA Pipelines Side by Side**
<br />


Another useful application of RCA is in enzyme development using random mutagenesis. Random mutagenesis is usually paired with genetic selection or high-throughput screening in order to intentionally replicate enzymes while introducing mutations. This is used to develop various forms of enzymes that can be used in new, innovative ways. The most common application of random mutagenesis is with an error-prone version of PCR, which is able to induce mutations in its replication process by weakening the accuracy of DNA Polymerase. While this is a common practice, it is tedious and requires many steps, including digestion with restriction enzymes, separation with electrophoresis, and ligation into a vector, all of which can sum up to over a day’s worth of processes. However, if we were to use an error-prone RCA method instead of PCR, the process becomes a lot less cumbersome for multiple reasons. Firstly, instead of specifically designed primers, RCA can use random hexamers as universal primers. Another significant advantage is that RCA does not require a thermal-cycler due to its isothermal nature. In addition, the end products of RCA are ready to be used for the direct formation of E. Coli, which is extremely useful in the subsequent steps of these types of experiments. One can easily see that RCA can act as a strong alternative to PCR (See Figure 3). 

<br />

![Figure 3 - RCA vs. PCR Pipelines for Error-Prone Amplification](https://github.com/npusarla/BENG183_Final_Projects_FALL2019/blob/master/Images/Figure3.jpeg)
&nbsp; &nbsp; &nbsp; &nbsp;**Figure 3 - RCA vs. PCR Pipelines for Error-Prone Amplification**
<br />


Rolling Circle Amplification is used frequently for efficient isothermal DNA amplification in molecular diagnoses. Even though PCR is a popular and reliable technique, its applicability is limited by its dependence on heating and cooling cycles. The heating step of PCR can damage biological models, and therefore an isothermal approach to DNA/RNA amplification, such as RCA, is preferable. The circular templates used allow for amplification to be over a thousand-fold from a single binding event, therefore, making RCA ideal for the ultrasensitive detection used in diagnostics. In addition, Rolling Circle Amplification, using the specific bacteriophage ɸ29, DNA Polymerase aids in determining diagnoses of geminiviruses and essentially all viruses with small-single stranded circular DNA genomes[2]. Viral DNA components of geminiviruses can be detected using RCA, allowing for such diagnoses. 

A specific example of this case is described below. In various tropical and subtropical countries around the world, geminiviruses have a history of destroying large regions of crops, and a combination of PCR and RFLP (Restriction Fragment Length Polymorphism) have been used for diagnostic tools in this field. However, developing countries are the major regions in which geminiviruses attack, and a combination of RCA/RFLP offers a low-cost procedure alternative for diagnosis that can be performed in lower standard laboratories, but still with high accuracy. Circular DNA was extracted from plants that were positive and negative for infection by various geminiviruses. After purification of the DNA, RCA was performed as an alternative to PCR using ɸ29 Polymerase. Through gel electrophoresis of various geminivirus infected plants, these amplified sequences showed clear differences between infected versus non-infected plants’ DNA. Further sequencing was then performed on this RCA-amplified DNA and through cycle-sequencing, 938 nucleotides were sequenced in one go with 99.5% fidelity. This shows that for this use-case, PCR and various other expensive cloning procedures are unnecessary, and RCA provides a cost-effective alternative[2].

The DNA amplified from RCA can additionally be characterized by RFLP analysis and directly sequenced up to 900 bases in a single run, avoiding other obstacles and allowing for quicker analysis. There are many other advantages to using RCA such as the lack of expensive devices, its simplicity, and the characterization of all infecting circular DNA components, thus, making it a widely used mechanism for diagnostics. 



### Citations 

1. Dean, Frank B. et al. “Rapid Amplification of Plasmid and Phage DNA Using Phi29 DNA Polymerase and Multiply-Primed Rolling Circle Amplification.” Genome Research 11.6 (2001): 1095–1099. 
2. Haible, Daniela, et al. “Rolling Circle Amplification Revolutionizes Diagnosis and Genomics of Geminiviruses.” Journal of Virological Methods, vol. 135, no. 1, July 2006, pp. 9–16., doi:10.1016/j.jviromet.2006.01.017. 
3. Ali, M. Monsur, et al. “Rolling Circle Amplification: a Versatile Tool for Chemical Biology, Materials Science and Medicine.” Chemical Society Reviews, The Royal Society of Chemistry, 18 Mar. 2014, pubs.rsc.org/en/content/articlelanding/2014/cs/c3cs60439j#!divAbstract. 
4. Fujii, Ryota & Kitaoka, Motomitsu & Hayashi, Kiyoshi. (2004). One-step random mutagenesis by error-prone rolling circle amplification. Nucleic acids research. 32. e145. 10.1093/nar/gnh147. 
5. Mohsen, Michael G, and Eric T Kool. “The Discovery of Rolling Circle Amplification and Rolling Circle Transcription.” Accounts of Chemical Research, U.S. National Library of Medicine, 15 Nov. 2016, www.ncbi.nlm.nih.gov/pubmed/27797171.





